File: src/patterns/creational/builder/car.nim
type
    Car* = ref object
        engine: string
        horsepower: uint16


proc newCar*(engine: string, horsepower: uint16): Car =
    return Car(
        engine: engine,
        horsepower: horsepower
    )

proc getCarInfo*(self: Car): string =
    return "Engine: " & self.engine & "\nHorsepower: " & $self.horsepower

proc getEngine*(self: Car): string =
    return self.engine

proc setEngine*(self: Car, engine: string) =
    self.engine = engine

proc getHorsepower*(self: Car): uint16 =
    return self.horsepower

proc setHorsepower*(self: Car, horsepower: uint16) =
    self.horsepower = horsepower

File: src/patterns/creational/builder/car.go
package builder


import(
	"strconv"
)


type Car struct {
	engine string
	horsepower uint16
}


func NewCar(engine string, horsepower uint16) *Car {
	return &Car{
		engine: engine,
		horsepower: horsepower,
	}
}

func (self *Car) GetCarInfo() string {
	return "Engine: " + self.engine + "\nHorsepower: " + strconv.FormatUint(uint64(self.horsepower), 10)
}

func (self *Car) GetEngine() string {
	return self.engine
}

func (self *Car) SetEngine(engine string) {
	self.engine = engine
}

func (self *Car) GetHorsepower() uint16 {
	return self.horsepower
}

func (self *Car) SetHorsepower(horsepower uint16) {
	self.horsepower = horsepower
}